### ISSUE TITLE
Obsolete check for libintl.h

### ISSUE THREAD

**Issue by Arfrever** [libpinyin/libpinyin#87]
`libintl.h` header is not included anywhere in this project, so checking for this header is needless.
`HAVE_LIBINTL_H` is also not used anywhere.

```
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -43,7 +43,6 @@
 
 include(CheckIncludeFileCXX)
 check_include_file_cxx(locale.h HAVE_LOCALE_H)
-check_include_file_cxx(libintl.h HAVE_LIBINTL_H)
 check_include_file_cxx(stdlib.h HAVE_STDLIB_H)
 check_include_file_cxx(string.h HAVE_STRING_H)
 check_include_file_cxx(sys/time.h HAVE_SYS_TIME_H)
--- configure.ac
+++ configure.ac
@@ -80,7 +80,7 @@
 AC_FUNC_MMAP
 AC_CHECK_FUNCS([gettimeofday memmove memset setlocale])
 
-AC_CHECK_HEADERS([libintl.h string.h])
+AC_CHECK_HEADERS([string.h])
 
 # Check Berkeley DB or Kyoto Cabinet
 DBM="BerkeleyDB"
```

By the way, if `string.h` was absent, then `AC_CHECK_HEADERS([string.h])` would just not define `HAVE_STRING_H`. However several source files include `string.h` unconditionally without checking `HAVE_STRING_H`.
So maybe `AC_CHECK_HEADERS([string.h], , AC_MSG_ERROR([string.h missing]))` should be used. See:
https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Generic-Headers.html
https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Printing-Messages.html
(Or drop check for `string.h` for simplicity since everybody should have this header...)

**Comment by epico** [comment]
Thanks for the report!

Fixed in https://github.com/epico/libpinyin/commit/86a756c0d888220b6c4e667642e0ee0ddb737d83 .

**Comment by Arfrever** [comment]
86a756c0d888220b6c4e667642e0ee0ddb737d83
Fix has been released in libpinyin 2.2.1.
